parsers/TR.py

```diff
@@ -100,23 +112,24 @@ def fetch_production(
 
     data = fetch_data(target_datetime=target_datetime, kind="production")
 
-    all_data_points = []
+    production_breakdowns = ProductionBreakdownList(logger)
     for item in data:
-        production = {}
-        for mode in PRODUCTION_MAPPING:
-            value = sum([item[key] for key in item if key in PRODUCTION_MAPPING[mode]])
-            production[mode] = round(value, 4)
-        data_point = {
-            "zoneKey": zone_key,
-            "datetime": _str_to_datetime(item.get("date")),
-            "production": production,
-            "source": "epias.com.tr",
-        }
-
-        all_data_points += [data_point]
+        mix = ProductionMix()
+        for key in item:
+            try:
+                mix.add_value(INVERT_PRODUCTION_MAPPPING[key], item[key])
+            except KeyError:
+                continue
```

Comment:
Was thinking about that, but wasn't sure whether to change the behaviour since previously any unknown keys went un-noticed!

Reply:
I notice that I get also `nucklear` and `wasteheat`, should I map them to something in this PR, ignore them, or leave it for a different PR?

Reply:
I'd say map them in this PR, if you wish. Otherwise create a follow-up pr.
But nuclear should be pretty straight forward at least and I don't think mapping waste heat to unknown will be an issue either.

And yeah we have a lot of old parsers that need a bit of fixing, personally I like to do that when making other changes anyway but if there are bigger changes it might make sense to do it in another PR.